### ISSUE TITLE
Content-security-policy and Form helper

### ISSUE THREAD

**Issue by MolbioUnige** [cakephp/cakephp#17729]
### Description

Is the inline style of the [hiddenBlock template](https://github.com/cakephp/cakephp/blob/2eecad68ca9d957f41a0cd2a76c6e111cb019d4e/src/View/Helper/FormHelper.php#L125) necessary? That messes with CSP, as one has to allow unsafe-inline in style-src, or use custom template to get rid of it.

**Comment by dereuromark** [comment]
It could just be a class, but then that class would need to be for sure injected into the CSS being rendered out, either in a separate file or in-page as CSS block on top.
If that could be managed out of the box for apps, then it would not be necessary as inline style.

Do you have a PR to propose such a change that works out of the box for apps?

**Comment by dereuromark** [comment]
https://github.com/cakephp/app/blob/9a7f511ed5f10dac993051cb1ecad79ab7cc93e5/templates/layout/default.php#L33
If we dynamically inject that template stuff into this block, it should work.
We just shouldn't inject it multiple times if the hidden block is used more than once.

**Comment by MolbioUnige** [comment]
No, sorry, I don't have anything to propose, as I don't understand why the div must be hidden.

**Comment by dereuromark** [comment]
So that hidden fields and _csrfField() etc are not visible or modifying the styling of the content.
If you don't use any of them, you should not be generating any `hiddenBlock` btw.

**Comment by MolbioUnige** [comment]
Ok, I understand now. I commented the csrf middleware while working on the csp middleware thus the hidden block only contained the hidden _method input. I completely forgot about the csrf field

**Comment by markstory** [comment]
The hidden block ensures that the hidden inputs stay hidden, and don't take up any visual space. If you're in a strict CSP environment, you could use custom form helper templates to override the template for `hiddenBlock` and use a classname or different element.

**Comment by MolbioUnige** [comment]
Yes, thanks, I did so

**Comment by ADmad** [comment]
FormHelper::postLink() also generates a form tag with inline style which is hardcoded and there's no convenient way to prevent.

**Comment by MolbioUnige** [comment]
[According to Mozilla](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/hidden), there is no way to make hidden fields visible in page content, so in that sens the surrounding div doesn't seem necessary.

I found a [12 years old discussion](https://www.sitepoint.com/community/t/why-put-input-hidden-field-in-display-none-style/19491/6) where someone said that, wrapping hidden fields in a display:none div, was done so as in legacy IE or older browser, hidden fields took up some space. But then, there might be other reasons.

**Comment by ADmad** [comment]
Sometimes CSS libs/devs add some default styling to all inputs which can potentially add spacing around hidden inputs too for e.g.

**Comment by markstory** [comment]
> Sometimes CSS libs/devs add some default styling to all inputs which can potentially add spacing around hidden inputs too for e.g.

This is the reason for the hidden div. Removing the hidden div could break existing applications so we'd have to wait for a major release to change it.

**Comment by dereuromark** [comment]
A side note: using postButton() can circumvent the issue ( https://github.com/dereuromark/cakephp-geo/pull/96 ) and allow stricter csp.

**Comment by dereuromark** [comment]
Also opened https://github.com/cakephp/cakephp/pull/19414 as more narrow enhancement and discussion.
